src/java/org/apache/cassandra/concurrent/SEPExecutor.java

```diff
@@ -260,23 +282,64 @@ public long getCompletedTaskCount()
 
     public int getActiveTaskCount()
     {
-        return maxWorkers - workPermits(permits.get());
+        return maxWorkers.get() - workPermits(permits.get());
+    }
+
+    public int getCorePoolSize()
+    {
+        return 0;
+    }
+
+    public void setCorePoolSize(int newCorePoolSize)
+    {
+        throw new IllegalArgumentException("Cannot resize core pool size of SEPExecutor");
     }
 
     @Override
     public int getMaximumPoolSize()
     {
-        return maxWorkers;
+        return maxWorkers.get();
+    }
+
+    @Override
+    public void setMaximumPoolSize(int newMaximumPoolSize)
+    {
+        final int oldMaxWorkers = maxWorkers.get();
+        long current;
+        int workPermits;
+
+        if (newMaximumPoolSize < 0)
+        {
+            throw new IllegalArgumentException("Maximum number of workers must not be negative");
+        }
+
+        int deltaWorkPermits = newMaximumPoolSize - oldMaxWorkers;
+        if (!maxWorkers.compareAndSet(oldMaxWorkers, newMaximumPoolSize))
+        {
+            throw new IllegalStateException("Maximum pool size has been changed while resizing");
+        }
+
+        if (deltaWorkPermits == 0)
+            return;
+
+        do
```

Comment:
Could avoid look with simply `permits.updateAndGet(cur -> updateWorkPermits(cur, workPermits(cur) + deltaWorkPermits)`

Reply:
thx, that cleans things up a bit.